x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -1872,6 +1873,7 @@ public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings unus
                 .setDescription("Contains ML configuration data")
                 .setMappings(MlConfigIndex.mapping())
                 .setSettings(MlConfigIndex.settings())
+                .setIndexFormat(new SystemIndexDescriptor.IndexFormat(0, "-622395039"))
```

Comment:
If no other changes are made to this PR then this hash will be invalid when the version on the main branch is bumped to 8.11.0. The reason is that the mappings include the version in their `_meta` field - see https://github.com/elastic/elasticsearch/blob/12ad399c488f0cc60e19b5e1b29c6d569cb4351a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlConfigIndex.java#L31-L37

So that implies that the version needs to be removed from the system index mappings before main is bumped to 8.11.0, otherwise the person doing the version bump is going to be faced with a load of test failures to fix.

However, removing version is also problematic, because for existing users doing a rolling upgrade there will be a period when some of the nodes in the cluster are running a version that thinks the version meta key should be present and others that rely on it having been removed.

A possible solution to that could be to remove the version meta key from the mappings before creating the hash.